docs/references/nuxt/clerk-middleware.mdx

```diff
@@ -64,53 +64,58 @@ export default clerkMiddleware((event) => {
+  const { has } = event.context.auth
+  const isInvoicesRoute = event.path.startsWith('/api/invoices')
+  const canCreateInvoices = has({
+    permission: 'org:invoices:create',
+  })
 
-  // Check if the user is trying to access a protected route
-  if (isAdminRoute) {
-    // Check if the user is signed in and has the required admin role
-    if (!userId || orgRole !== 'org:admin') {
+  // Check if user is accessing sensitive customer data
+  if (isInvoicesRoute) {
+    // Check if user has the required permission
+    if (!canCreateInvoices) {
       throw createError({
         statusCode: 403,
-        statusMessage: 'Forbidden: Admin access required',
+        statusMessage: 'Unauthorized: Missing permission to create invoices',
       })
     }
   }
 })
 ```
 
-#### Protecting routes using custom permissions
+#### Protecting routes using default roles
 
-System permissions (e.g., `org:sys_domains:manage`) aren't included in session claims and can't be used with `auth.has()` on the server-side. To perform permission checks in middleware, **you must create and use custom permissions**. See the [dedicated guide](/docs/organizations/roles-permissions#permissions).
+To protect a route based on a default role, you can check the `orgRole` property on the [`auth`](/docs/references/nuxt/overview#auth-object) object.
 
-In the following example, the `clerkMiddleware()` helper verifies if the user is accessing an invoices route. If so, it checks if the user has the required custom permission. If not, an error is thrown using Nuxt's [`createError`](https://nuxt.com/docs/api/utils/create-error) utility.
+> [!WARNING]
+> It's best practice to use permission-based authorization over role-based authorization, as it reduces complexity and increases security. Usually, complex role checks can be refactored with a single permission check.
+
+In the following example, the `clerkMiddleware()` helper checks if the user is accessing a protected route. If so, it checks the `orgRole` property to ensure the user has the required admin role. If they don't, an error is thrown using Nuxt's [`createError()`](https://nuxt.com/docs/api/utils/create-error) utility.
 
 ```ts {{ filename: 'server/middleware/clerk.ts' }}
 import { clerkMiddleware } from '@clerk/nuxt/server'
 
 export default clerkMiddleware((event) => {
-  const { has } = event.context.auth
-  const isInvoicesRoute = event.path.startsWith('/api/invoices')
-  const canCreateInvoices = has({
-    permission: 'org:invoices:create',
-  })
+  const { orgRole } = event.context.auth
+  const isAdminRoute = event.path.startsWith('/api/admin')
 
-  // Check if user is accessing sensitive customer data
-  if (isInvoicesRoute) {
-    // Check if user has the required permission
-    if (!canCreateInvoices) {
+  // Check if the user is trying to access a protected route
+  if (isAdminRoute) {
+    // Check if the user has the required admin role
+    if (orgRole !== 'org:admin') {
```

Comment:
I think we can also use  the `role` property here @alexisintech? Something lik,e `!has({ role: 'org:admin' })`

Reply:
yes, either orgRole or has({ role }). I can add an example

Reply:
[c66fd2d](https://github.com/clerk/clerk-docs/pull/1728/commits/c66fd2d10ca2736b4c5b3c25eac93da8c4775c50)